tests/actor_pool/test_actor_context.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import os
+from contextlib import contextmanager
+
+import pytest
+import ray
+
+import daft
+from daft import ResourceRequest
+from daft.context import get_actor_context, get_context, set_planning_config
+from daft.datatype import DataType
+from daft.internal.gpu import cuda_visible_devices
+from daft.udf import udf
+
+pytestmark = pytest.mark.skipif(
+    get_context().daft_execution_config.enable_native_executor is True,
+    reason="Native executor fails for these tests",
+)
+
+
+@pytest.fixture(scope="module")
+def enable_actor_pool():
+    try:
+        original_config = get_context().daft_planning_config
+
+        set_planning_config(
+            config=get_context().daft_planning_config.with_config_values(enable_actor_pool_projections=True)
+        )
+        yield
+    finally:
+        set_planning_config(config=original_config)
+
+
+@contextmanager
+def reset_runner_with_gpus(num_gpus, monkeypatch):
+    """If current runner does not have enough GPUs, create a new runner with mocked GPU resources"""
+    if len(cuda_visible_devices()) < num_gpus:
+        if get_context().runner_config.name == "ray":
+            try:
+                ray.shutdown()
+                ray.init(num_gpus=num_gpus)
+                yield
+            finally:
+                ray.shutdown()
+                ray.init()
+        else:
+            try:
+                monkeypatch.setenv("CUDA_VISIBLE_DEVICES", ",".join(str(i) for i in range(num_gpus)))
```

Comment:
Other useful monkeypatch corner-cases are when `CUDA_VISIBLE_DEVICES` isn't set, or if it is set to something invalid etc